gptp: Add monotonic raw clock option

christopher-s-hall opened the pull request:
Timestamp clocks that implement the PTP_SYS_OFFSET_PRECISE ioctl return
 cross-timestamps with respect to monotonic raw and realtime
Add configuration file option allowing clock selection for system/local
 clock relation

The default clock type is Realtime which is equivalent to the
 CLOCK_REALTIME option to clock_gettime
Optionally, MonotonicRaw (equivalent to CLOCK_MONOTONIC_RAW) can be
 specified in the gPTP daemon configuration file:

[clock]

SystemClock = MonotonicRaw

Fixed Makefile issue where some shell configurations use built-in 'echo'
 and others don't
Forces all to use /bin/echo

andrew-elder commented on daemons/gptp/gptp_cfg.ini:
```diff
@@ -6,6 +6,11 @@
 # The lower the number, the higher the priority for the BMCA.
 priority1 = 248
 
+# Option to specify system clock type
```
Can we add some more details here. Specifically, we should list the currently supported options, the default option and what the options mean. The text that is part of the pull request has great information in it, but I think that 6 months from now readers will prefer to read an updated comment here.

pinealservo replied:
Yes, I found comments in the example ini files very helpful when trying to actually use the avtp_pipeline code, so it'd be a nice help for users.

andrew-elder commented:
Looks good to me. One small comment update requested. Can anyone else review this?

pinealservo commented on daemons/gptp/common/gptp_cfg.hpp:
```diff
@@ -68,6 +69,9 @@ class GptpIniParser
             /*ptp data set*/
             unsigned char priority1;
 
+		/* Clock data set */
```
It seems like there might be a tabs vs. spaces conflict in this file; I'm not sure whether it'd be better to just make the change match or if the file itself has a mismatched convention.

christopher-s-hall commented:
I added more comments to the config file and I added a second commit to fix the mixed whitespace in gptp_cfg.hpp at least for that struct. My spacing was/is correct (tabs only). What was there wasn't.

andrew-elder commented:
@pinealservo - looks good to me. If you agree, please go ahead and merge. Thanks for the updates @christopher-s-hall
